pkg/etcd/etcd.go

```diff
@@ -853,10 +861,152 @@ func (e *ETCD) Snapshot(ctx context.Context, config *config.Control) error {
 		}
 	}
 
+	snapshots, err := e.listSnapshots(ctx, snapshotDir)
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("Saving current etcd snapshot set to %s ConfigMap", snapshotConfigMapName)
+	return e.storeSnapshotData(ctx, snapshots)
+}
+
+type s3Config struct {
```

Comment:
@ibuildthecloud are these somewhere that we could reference in rancher/rancher, or do we just need to maintain two copies of the struct definition?